jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -191,18 +194,49 @@ public JibContainerBuilder createJibContainerBuilder(
 
       FileCollection snapshotDependencies =
           nonProjectDependencies.filter(file -> file.getName().contains("SNAPSHOT"));
+
+      List<String> changingCollections =
+          runtimeConfiguration
+              .getAllDependencies()
+              .stream()
+              .filter(dependency -> ((ExternalModuleDependency) dependency).isChanging())
+              .map(
+                  dependency ->
+                      dependency
+                          .getName()
+                          .concat("-")
+                          .concat(dependency.getVersion())
+                          .concat(".jar"))
+              .collect(Collectors.toList());
+
+      List<Path> changingDependenciesCollection =
+          nonProjectDependencies
+              .getFiles()
+              .stream()
+              .filter(
+                  file ->
+                      changingCollections != null && changingCollections.contains(file.getName()))
```

Comment:
null check seems unnecessary, the list may be empty, but I don't think `.collect(Collectors.toList())` ever returns null.